### PR TITLE
Bring Kafka operator up to best practices

### PR DESCRIPTION
**Pull request by lfrancke**
- License -> OSL 3
- Sort Cargo.toml, add license & description
- Move CRD YAML file to deploy/crd
- Cargo update

### CODE REVIEW

**Review by maltesander — APPROVED** [review]
Looks good.

File: server/src/bin/generate_kafka_crd.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+use stackable_kafka_crd::KafkaCluster;
+use std::error::Error;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    println!("{}", serde_yaml::to_string(&KafkaCluster::crd())?);
+    Ok(())
+}
```
Comment (maltesander) [inline server/src/bin/generate_kafka_crd.rs:7]: Can you check with crd/src/main.rs. There the crd is generated and written (not to the deploy folder).
If you want the console output i would suggest to merge these two.

**Review by maltesander — CHANGES_REQUESTED** [review]
We have two places where we write or print the CRD now. I understand you want to have it within the "server", but i still like the mechanic to directly write into the file. What do you think?

File: server/src/bin/generate_kafka_crd.rs (hunk 1 above)
Comment (lfrancke) [reply server/src/bin/generate_kafka_crd.rs:7]: The crd/src/main will also be called manually, right?
Either way: It's not ideal but I'd suggest we merge as is for now until we can finally get a generic way to do so in kube-rs.

Comment (maltesander) [reply server/src/bin/generate_kafka_crd.rs:7]: Yes called manually but writes into the files. So one step less ;)

**Review by maltesander — APPROVED** [review]

**Pull request merged**